common/chunkStatusLogger.go

```diff
@@ -315,6 +319,26 @@ func (csl *chunkStatusLogger) FlushLog() {
 	}
 }
 
+// CloseLogger close the chunklogger thread.
+func (csl *chunkStatusLogger) CloseLogger() {
+	// Once logger is closed, we log no more chunks.
+	csl.outputEnabled = false
+
+	// lets make sure no previous entry exist in channel.
+	for {
+		if len(csl.unsavedEntries) == 0 {
+			break
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+
+	// Signal main logger goroutine to packup.
+	csl.done <- struct{}{}
```

Comment:
As discussed, we don't need this. Closing of the channel is enough indication to the logger to mean "end processing"